shipyard-controller/db/mongodb_staterepo_test.go

```diff
@@ -0,0 +1,180 @@
+package db_test
+
+import (
+	"context"
+	"github.com/keptn/keptn/shipyard-controller/db"
+	"github.com/keptn/keptn/shipyard-controller/models"
+	"github.com/ory/dockertest/v3"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"os"
+	"testing"
+)
+
+func setupLocalMongoDB() (*dockertest.Pool, *dockertest.Resource) {
+	os.Setenv("MONGODB_HOST", "localhost:27017")
+	os.Setenv("MONGO_DB_NAME", "keptn")
+	os.Setenv("MONGODB_USER", "keptn")
+	os.Setenv("MONGODB_PASSWORD", "password")
+
+	// uses a sensible default on windows (tcp/http) and linux/osx (socket)
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		log.Fatalf("Could not connect to docker: %s", err)
+	}
+	var mongoClient *mongo.Client
+	// pulls an image, creates a container based on it and runs it
+	resource, err := pool.Run("docker.io/centos/mongodb-36-centos7", "1", []string{"MONGODB_DATABASE=keptn", "MONGODB_PASSWORD=password", "MONGODB_USER=keptn", "MONGODB_ADMIN_PASSWORD=password"})
```

Comment:
Can we please also evaluate whether using testcontainers-go is a better solution?
https://github.com/testcontainers/testcontainers-go

Also, please let's make sure this line is formatted properly (e.g., newlines vor environment variables)

Reply:
https://github.com/ory/dockertest seems to be more actively maintained, and has more stars on GitHub (1.8k vs 860) - hence I decided to go with that solution

Reply:
Roger, let's go with dockertest.